core/src/main/java/org/apache/brooklyn/core/catalog/internal/BasicBrooklynCatalog.java

```diff
@@ -859,6 +869,34 @@ private String makeAsIndentedObject(String yaml) {
         log.debug("Adding manual catalog item to "+mgmt+": "+yaml);
         checkNotNull(yaml, "yaml");
         List<CatalogItemDtoAbstract<?, ?>> result = collectCatalogItems(yaml);
+
+        // Iterate over the results to add the interfaces tag. We are doing this here to be able to get the backward-types.
+        for (final CatalogItemDtoAbstract<?, ?> item : result) {
+            if (item.getCatalogItemType().equals(CatalogItemType.ENTITY) || item.getCatalogItemType().equals(CatalogItemType.TEMPLATE)) {
+                Class<?> type = CatalogUtils.newClassLoadingContext(mgmt, item).tryLoadClass(item.getReferenceType()).orNull();
```

Comment:
I think it would be easier to create a spec out of the catalog item and get the type from there. What if referenceType is another catalog item.